library/cc/engine_builder.h

```diff
@@ -46,6 +47,9 @@ class EngineBuilder {
   // Filter): EngineBuilder& addNativeFilter(name: String = UUID.randomUUID().toString(),
   // typedConfig: String): EngineBuilder& addStringAccessor(name: String, accessor:
   // EnvoyStringAccessor): EngineBuilder {
+protected:
+  void setOverrideConfigForTests(std::string config) { config_override_for_tests_ = config; }
+  void setAdminAddressPathForTests(std::string admin) { admin_address_path_for_tests_ = admin; }
```

Comment:
alternately could publicly add key-value and hash map for generic command line override.